UPGRADE-2.8.md

```diff
@@ -42,6 +42,194 @@ Form
        private $author;
    }
    ```
+   
+ * Type names were deprecated and will be removed in Symfony 3.0. Instead of
+   referencing types by name, you should reference them by their
+   fully-qualified class name (FQCN) instead. With PHP 5.5 or later, you can
+   use the "class" constant for that:
+   
+   Before:
+   
+   ```php
+   $form = $this->createFormBuilder()
+       ->add('name', 'text')
+       ->add('age', 'integer')
+       ->getForm();
+   ```
+   
+   After:
+   
+   ```php
+   use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+   use Symfony\Component\Form\Extension\Core\Type\TextType;
+   
+   $form = $this->createFormBuilder()
+       ->add('name', TextType::class)
+       ->add('age', IntegerType::class)
+       ->getForm();
+   ```
+   
+   As a further consequence, the method `FormTypeInterface::getName()` was
+   deprecated and will be removed in Symfony 3.0. You should remove this method
+   from your form types.
+   
+   If you want to customize the block prefix of a type in Twig, you should now
+   implement `FormTypeInterface::getBlockPrefix()` instead:
```

Comment:
What exactly does "block prefix of a type in Twig" mean?

I'm upgrading old Symfony projects and this line is a mystery for me. Can I remove the `getName()` or not?
I'd appreciated some Twig snippet example what exactly is meant here.

Reply:
This refers to how form fragment names are generated. By default, the value is derived from the class name (see e.g. https://symfony.com/doc/current/form/form_themes.html#form-fragment-naming where `textarea` is derived from `TextareaType` class). If you have your own `TextareaType` class, you will maybe want to override the `getBlockPrefix()` method to not reuse the blocks used by the core `TextareaType`.

Reply:
I never used such fragments and docs is very confusing for me. I don't see any twig sample for `textarea_widget` keyword. I also checked PRs referecing this PR when upgrading, but it seems nobody ever used it on custom types. 

I look for 3 lines of twig sample, nothing more. So I just can scan twig templates with regualar expression. There are no Twig tests that could tell us what broke.


```twig
{{ form_widget(form.age) }}

{{ textarea_widget(form.age) }}
```

So for custom:

```php
class SomeType extends FormType
{
	public function getName()
    {
    	return 'yolo';
	}
}
```

↓

```twig
{{ yolor_widget(form.age) }}
```

Like this?

So it's string from `getName()` + somehow `_widget` magically added?

Reply:
If you just looking for a way to automatically update the code of the form type class, I would compare the value returned in `getName()` with what the automatically computed name would be (see https://github.com/symfony/symfony/blob/ba4d57bb5fc0e9a1b4f63ced66156296dea3687e/src/Symfony/Component/Form/AbstractType.php#L55). Only if both values differ, you will need to rename the `getName()` method to `getBlockPrefix()`. Otherwise it can be removed.

Reply:
That's what I already do with https://github.com/rectorphp/rector/pull/3705 I made it just before reading your comment :D

I want to take it further and remove all the `getBlockPrefix()` that don't have standard name,
and are never used in the twig templates. That's why I ask how can I find it in the twig templates.

Reply:
The special block names are `*_row`, `*_label`, `*_widget` and `*_errors` (nowadays there is also `*_help`, but I guess you are addressing 2.8 applications here). But you need to be careful as the block prefix is also used to build block names for `CollectionType` entries (see https://symfony.com/doc/2.8/form/form_customization.html#form-custom-prototype).